### PR TITLE
DOCSP-39405: kotlin v5.1 content

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-kotlin/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-39405>
Staging:
- [oidc](https://preview-mongodbrustagir.gatsbyjs.io/kotlin/DOCSP-39405-v5.1-content/fundamentals/enterprise-auth/#std-label-kotlin-oidc)
- [connection options](https://preview-mongodbrustagir.gatsbyjs.io/kotlin/DOCSP-39405-v5.1-content/fundamentals/connection/connection-options/#std-label-connection-options:~:text=serverMonitoringMode)
- [whats new](https://preview-mongodbrustagir.gatsbyjs.io/kotlin/DOCSP-39405-v5.1-content/whats-new/#std-label-kotlin-coroutine-version-5.1)

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/fundamentals/enterprise-auth.txt [hunk 1]
```diff
@@ -278,4 +287,131 @@ mechanism:
...
+
+MONGODB-OIDC
+~~~~~~~~~~~~
+
+.. important::
+
+   The MONGODB-OIDC authentication mechanism requires {+mdb-server+} v7.0 or later running
+   on a Linux platform.
+
+The following sections describe how to use the ``MONGODB-OIDC``
```
Comment (norareidy) [inline source/fundamentals/enterprise-auth.txt:301]: S: I don't think this needs to be monospaced, based on the other mentions of MONGODB-OIDC, Kerberos, LDAP
```suggestion
The following sections describe how to use the MONGODB-OIDC
```

**Review by norareidy — APPROVED** [review]
LGTM + a few suggestions!

File: examples/src/test/kotlin/EnterpriseAuthTest.kt [hunk 2]
```diff
@@ -113,6 +117,88 @@ internal class EnterpriseAuthTest {
...
+                OidcCallbackResult(accessToken)
+            }
+        // :snippet-end:
+    }
+
+
+    fun oidcCallbackFile() = runBlocking {
+        // :snippet-start: oidc-callback-file
+        val credential = MongoCredential.createOidcCredential(null)
+            .withMechanismProperty("OIDC_CALLBACK") { Context context ->
```
Comment (rozza) [inline examples/src/test/kotlin/EnterpriseAuthTest.kt:187]: This doesn't compile in Kotlin try: 

```suggestion
            .withMechanismProperty("OIDC_CALLBACK") { context: Context ->
```

File: examples/src/test/kotlin/EnterpriseAuthTest.kt [hunk 3]
```diff
@@ -113,6 +117,88 @@ internal class EnterpriseAuthTest {
...
+                }
+                .credential(credential)
+                .build())
+        // :snippet-end:
+    }
+
+    fun oidcCallback() = runBlocking {
+        // :snippet-start: oidc-callback
+        val credential = MongoCredential.createOidcCredential(null)
+            .withMechanismProperty("OIDC_CALLBACK") { Context context ->
```
Comment (rozza) [inline examples/src/test/kotlin/EnterpriseAuthTest.kt:176]: This doesn't compile in Kotlin try:

```suggestion
            .withMechanismProperty("OIDC_CALLBACK") { context: Context ->
```

**Review by rozza — CHANGES_REQUESTED** [review]
Just a kotlin bug then LGTM

**Pull request merged**